L1Trigger/Phase2L1ParticleFlow/plugins/L1TCorrelatorLayer1Producer.cc

```diff
@@ -478,36 +531,62 @@ void L1TCorrelatorLayer1Producer::produce(edm::Event &iEvent, const edm::EventSe
     addMuon(mu, l1t::PFCandidate::MuonRef(muons, i));
   }
   // ------ READ CALOS -----
-  edm::Handle<l1t::PFClusterCollection> caloHandle;
-  for (const auto &tag : emCands_) {
-    iEvent.getByToken(tag, caloHandle);
-    const auto &calos = *caloHandle;
-    for (unsigned int ic = 0, nc = calos.size(); ic < nc; ++ic) {
-      const auto &calo = calos[ic];
-      if (debugR_ > 0 && deltaR(calo.eta(), calo.phi(), debugEta_, debugPhi_) > debugR_)
-        continue;
-      if (calo.pt() > emPtCut_)
-        addEmCalo(calo, l1t::PFClusterRef(caloHandle, ic));
+
+  // this is for parsing raw calo information
+  if (!emGCTRawCands_.isUninitialized()) {
+    auto caloHandle = iEvent.getHandle(emGCTRawCands_);
+    const auto &links = *caloHandle;
+    for (unsigned int ic = 0; ic < links.size(); ++ic) {
+      const auto &link = links[ic];
+      for (unsigned int ie = 0; ie < link.size(); ++ie) {
+        addGCTEmCaloRaw(link, ic, ie);
+      }
     }
   }
-  for (const auto &tag : hadCands_) {
-    iEvent.getByToken(tag, caloHandle);
-    const auto &calos = *caloHandle;
-    for (unsigned int ic = 0, nc = calos.size(); ic < nc; ++ic) {
-      const auto &calo = calos[ic];
-      if (debugR_ > 0 && deltaR(calo.eta(), calo.phi(), debugEta_, debugPhi_) > debugR_)
-        continue;
-      if (calo.pt() > hadPtCut_)
-        addHadCalo(calo, l1t::PFClusterRef(caloHandle, ic));
+
+  if (!hadHGCalCands_.isUninitialized()) {
+    auto caloHandle = iEvent.getHandle(hadHGCalCands_);
+    if (caloHandle.isValid()) {
+      const auto &calos = *caloHandle;
+      for (unsigned int ic = 0, nc = calos.size(); ic < nc; ++ic) {
```

Comment:
Why define `nc` here? Also here: https://github.com/cms-sw/cmssw/blob/d29d4100ada0be4b96ae53eb1085eae4bb1ae7cc/L1Trigger/Phase2L1ParticleFlow/plugins/L1TCorrelatorLayer1Producer.cc#L573

Reply:
This is a relatively common idiom to cast `size_t` explicitly. Anyway, I see no problem with caching the size here, no?

Reply:
I was just curious why it wasn't just:
`for (unsigned int ic = 0; ic < calos.size(); ++ic) {`
or even something like:
`for (auto ic : calos) {`
?